From fca79466ccc32b5f192197cb8e6667c273bb8a9d Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Tue, 21 Sep 2021 19:33:44 +0300
Subject: [PATCH 06/24] s32g: Move BL32_BASE to avoid overlapping with BL33

Changed the S32G_BL32_BASE define to be based on the base
address of the BL33, instead of the BL31. Otherwise, an
overlapping between BL32 and BL33 could occur.

Issue: ALB-7733
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32g/include/s32g_platform_def.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plat/nxp/s32g/include/s32g_platform_def.h b/plat/nxp/s32g/include/s32g_platform_def.h
index acb6e0dd0..80a6c5db8 100644
--- a/plat/nxp/s32g/include/s32g_platform_def.h
+++ b/plat/nxp/s32g/include/s32g_platform_def.h
@@ -188,8 +188,8 @@
  * these values can be further shrunk. The current values are preliminary.
  */
 #define S32G_BL32_SIZE		0x01600000
-#define S32G_BL32_BASE		(BL31_BASE - S32G_BL32_SIZE)
-#define S32G_BL32_LIMIT		(BL31_BASE)
+#define S32G_BL32_BASE		(BL33_BASE - S32G_BL32_SIZE)
+#define S32G_BL32_LIMIT		(BL33_BASE)
 
 #define FIP_BASE		(S32G_SRAM_END - FIP_MAXIMUM_SIZE)
 
-- 
2.17.1

